asv_bench/benchmarks/omnisci/benchmarks.py

```diff
@@ -238,19 +238,19 @@ def setup(self, value_type, shape, limit):
         self.df = pd.DataFrame(np.nan, index=pd.RangeIndex(shape[0]), columns=columns)
         trigger_import(self.df)
 
-        value = self.create_fillna_value(value_type, shape)
+        value = self.create_fillna_value(value_type, columns)
         limit = int(limit * shape[0]) if limit else None
         self.kw = {"value": value, "limit": limit}
 
     def time_fillna(self, value_type, shape, limit):
         execute(self.df.fillna(**self.kw))
 
     @staticmethod
-    def create_fillna_value(value_type: str, shape: tuple):
+    def create_fillna_value(value_type: str, columns: list):
         if value_type == "scalar":
             value = 18.19
         elif value_type == "dict":
-            value = {k: k * 1.23 for k in range(shape[0])}
+            value = {k: i * 1.23 for i, k in enumerate(columns)}
```

Comment:
are you going to do the same with ray-engine benchmarks?

Reply:
yes, https://github.com/modin-project/modin/issues/3246#issuecomment-881278686

Reply:
why do we need to split this into two separate PRs if the problems are identical? I think it would be logical to fix misbehavior for both of the engines in this PR and deal with the displaying problem in the separate one

Reply:
The code will look complicated and I want to split the functionality into several classes for simplicity. Maybe some more changes are needed to simplify the display in the dashboard, which looks like a problem for a separate tracker.
```python
        elif value_type == "dict":
            if self_type == "DataFrames":
                self.value = {k: i * 1.23 for i, k in enumerate(columns)}
            if self_type == "Series":
                self.value = {k: k * 1.23 for k in range(shape[0])}
        elif value_type == "Series":
            if self_type == "DataFrames":
                self.value = pd.Series(
                    [i * 1.23 for i in range(len(columns))], index=columns
                )
            if self_type == "Series":
                self.value = pd.Series(
                    [k * 1.23 for k in range(shape[0])], index=pd.RangeIndex(shape[0])
                )
```

Reply:
ok